homeassistant/components/evohome/__init__.py

```diff
@@ -135,14 +141,80 @@ def _handle_exception(err) -> bool:
             )
             return False
 
-        raise  # we don't expect/handle any other ClientResponseError
+        raise  # we don't expect/handle any other Exceptions
 
 
 async def async_setup(hass: HomeAssistantType, config: ConfigType) -> bool:
     """Create a (EMEA/EU-based) Honeywell evohome system."""
-    broker = EvoBroker(hass, config[DOMAIN])
-    if not await broker.init_client():
+
+    def load_auth_tokens(app_storage) -> Tuple[Dict, Optional[Dict]]:
+        tokens = dict(app_storage if app_storage else {})
+
+        if tokens.pop(CONF_USERNAME, None) != config[DOMAIN][CONF_USERNAME]:
+            return ({}, None)  # any tokens wont be valid
+
+        # evohomeasync2 requires naive/local datetimes as strings
+        if tokens[ACCESS_TOKEN_EXPIRES] is not None:
+            tokens[ACCESS_TOKEN_EXPIRES] = _dt_to_local_naive(
+                dt_util.parse_datetime(tokens[ACCESS_TOKEN_EXPIRES])
+            )
+
+        session_data = tokens.pop(USER_DATA, None)
+
+        return (tokens, session_data)
+
+    store = hass.helpers.storage.Store(STORAGE_VERSION, STORAGE_KEY)
+    app_storage = await store.async_load()
+
+    tokens, user_data = load_auth_tokens(app_storage)
+
+    client_v2 = evohomeasync2.EvohomeClient(
+        config[DOMAIN][CONF_USERNAME],
+        config[DOMAIN][CONF_PASSWORD],
+        **tokens,
+        session=async_get_clientsession(hass),
+    )
+
+    try:
+        await client_v2.login()
+    except (aiohttp.ClientError, evohomeasync2.AuthenticationError) as err:
+        _handle_exception(err)
         return False
+    finally:
+        config[DOMAIN][CONF_PASSWORD] = "REDACTED"
+
+    loc_idx = config[DOMAIN][CONF_LOCATION_IDX]
+    try:
+        loc_config = client_v2.installation_info[loc_idx][GWS][0][TCS][0]
+    except IndexError:
+        _LOGGER.error(
+            "Config error: '%s' = %s, but the valid range is 0-%s. "
+            "Unable to continue. Fix any configuration errors and restart HA.",
+            CONF_LOCATION_IDX,
+            loc_idx,
+            len(client_v2.installation_info) - 1,
+        )
+        return False
+
+    _LOGGER.debug("Config = %s", loc_config)
+
+    if config[DOMAIN][CONF_HIGH_PRECISION]:
+        client_v1 = evohomeasync.EvohomeClient(
+            client_v2.username,
+            client_v2.password,
+            user_data=user_data,
+            session=async_get_clientsession(hass),
+        )
+    else:
+        client_v1 = None
+
+    hass.data[DOMAIN] = {}
+    hass.data[DOMAIN]["broker"] = broker = EvoBroker(
+        hass, client_v2, client_v1, store, config[DOMAIN]
+    )
+
+    if _LOGGER.isEnabledFor(logging.DEBUG):  # don't do an I/O unless required
+        await broker.update()  # includes: _LOGGER.debug("Status = %s"...
```

Comment:
Are we deciding doing an update on logging setting? What's happening here?

Reply:
> Are we deciding doing an update on logging setting? 

No.

> What's happening here?

Due to the way the underlying client works, there is no valid way to get the (useful) logging data without calling a client library method (with a subsequent I/O) that happens to be the same method used for an update.